### PR TITLE
Issue: #2621870 by neetu. Correcting the code as per D8 latest standard. 

### PR DESCRIPTION
**Pull request by neetumorwani**
This is WIP. Not to be merged.

### CODE REVIEW

**Comment by ajitdev** [comment]
@neetumorwani Thanks! I will review when you are done with the changes.
Could you also please create an issue in the module's issue queue and explain the changes that have been done? In this way, others who are working on this module would know the progress and could help. This PR should be referenced from the issue.

File: autologout.module [hunk 1]
```diff
@@ -359,6 +370,7 @@ function _autologout_logout() {
 
   if (\Drupal::config('autologout.settings')->get('autologout_use_watchdog')) {
     \Drupal::logger('user')->info('Session automatically closed for %name by autologout.', array('%name' => $user->getUsername()));
```
Comment (fly2abhishek) [inline autologout.module:370]: This should be changed too.

```
if (Drupal::config('autologout.settings')->get('autologout_use_watchdog')) {
     Drupal::logger('user')->info('Session automatically closed for %name by autologout.', array('%name' => $user->getUsername()));
```

File: autologout.module [hunk 2]
```diff
@@ -64,8 +70,9 @@ function autologout_timeout_validate($timeout, $max_timeout = NULL) {
 /**
  * Adds a field to user/edit to change that users logout.
  */
-function autologout_form_user_form_alter(&$form, FormStateInterface $form_state) {
-  $user = \Drupal::currentUser();
+function autologout_form_user_form_alter(&$form, $form_state) {
```
Comment (ajitdev) [inline autologout.module:73]: hook_form_alter takes in FormStateInterface as a second argument
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Form%21form.api.php/function/hook_form_alter/8
Why was this removed?

File: autologout.services.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+services:
+  autologout_event_subscriber:
+    class: Drupal\autologout\EventSubscriber\AutologoutSubscriber
+    tags:
+      - {name: event_subscriber}
```
Comment (ajitdev) [inline autologout.services.yml:5]: Newline should be added here.

File: js/autologout.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 (function ($) {
 
   Drupal.behaviors.autologout = {
-    attach: function(context, settings) {
+    attach: function(context, settings,drupalSettings) {
```
Comment (ajitdev) [inline js/autologout.js:4]: Space after comma.

File: autologout.module [hunk 5]
```diff
@@ -50,7 +56,7 @@ function autologout_help($path, $arg) {
 function autologout_timeout_validate($timeout, $max_timeout = NULL) {
   $validate = TRUE;
   if (is_null($max_timeout)) {
-    $max_timeout = \Drupal::config('autologout.settings')->get('autologout_max_timeout');
+    $max_timeout = Drupal::config('autologout.settings')->get('autologout_max_timeout');
```
Comment (ajitdev) [inline autologout.module:59]: Can you revert this. Could be cleaned up in a separate commit.

File: autologout.module [hunk 6]
```diff
@@ -64,8 +70,9 @@ function autologout_timeout_validate($timeout, $max_timeout = NULL) {
 /**
  * Adds a field to user/edit to change that users logout.
  */
-function autologout_form_user_form_alter(&$form, FormStateInterface $form_state) {
-  $user = \Drupal::currentUser();
+function autologout_form_user_form_alter(&$form, $form_state) {
+  $user_timeout = _autologout_get_user_timeout();
+  $user = Drupal::currentUser();
```
Comment (ajitdev) [inline autologout.module:75]: Can you revert this. Could be cleaned up in a separate commit.
Use `\Drupal` for now instead of just `Drupal`.

File: autologout.module [hunk 7]
```diff
@@ -80,7 +87,7 @@ function autologout_form_user_form_alter(&$form, FormStateInterface $form_state)
     $form['autologout_user_' . $userid] = array(
       '#type' => 'textfield',
       '#title' => t('Your current logout threshold'),
-      '#default_value' => \Drupal::config('autologout.settings')->get('autologout_user_' . $userid),
+      '#default_value' => Drupal::config('autologout.settings')->get('autologout_user_' . $userid),
```
Comment (ajitdev) [inline autologout.module:90]: Same. Please revert.

File: autologout.module [hunk 8]
```diff
@@ -130,7 +137,7 @@ function autologout_user_profile_submit(&$form, FormStateInterface $form_state)
  * Implements hook_autologout_prevent().
  */
 function autologout_autologout_prevent() {
-  $user = \Drupal::currentUser();
+  $user = Drupal::currentUser();
```
Comment (ajitdev) [inline autologout.module:140]: Same. Please revert.

File: autologout.module [hunk 9]
```diff
@@ -139,8 +146,9 @@ function autologout_autologout_prevent() {
     'autologout_ahah_logout',
     'autologout_ahah_set_last',
   );
-
-  if (in_array(arg(0), $paths)) {
+  $url = Url::fromRoute('<current>');
+  $path_args = explode('/',$url->toString());
```
Comment (ajitdev) [inline autologout.module:150]: Space after the comma.

File: autologout.module [hunk 10]
```diff
@@ -160,8 +168,9 @@ function autologout_autologout_prevent() {
  * Implements hook_autologout_refresh_only().
  */
 function autologout_autologout_refresh_only() {
-  // Check to see if an open admin page will keep login alive.
-  if (!\Drupal::config('autologout.settings')->get('autologout_enforce_admin') && path_is_admin(current_path())) {
+  $url = Url::fromRoute('<current>');
+  $route = explode('/',$url->toString());
```
Comment (ajitdev) [inline autologout.module:172]: Space after the comma.

File: autologout.module [hunk 11]
```diff
@@ -160,8 +168,9 @@ function autologout_autologout_prevent() {
...
+  if (!Drupal::config('autologout.settings')->get('autologout_enforce_admin') && \Drupal::service('router.admin_context')->isAdminRoute(\Drupal::routeMatch()->getRouteObject())) {
```
Comment (ajitdev) [inline autologout.module:173]: Can you get the values in a variable first and then make the comparison?
Would make it more readable.

File: autologout.module [hunk 12]
```diff
@@ -177,7 +186,7 @@ function autologout_autologout_refresh_only() {
  * loaded from cache and an autologout timeout refresh needs to be triggered.
  */
 function autologout_page_build(&$page) {
-  $user = \Drupal::currentUser();
+  $user = Drupal::currentUser();
```
Comment (ajitdev) [inline autologout.module:189]: Same. Please revert.

File: autologout.module [hunk 13]
```diff
@@ -196,19 +205,19 @@ function autologout_page_build(&$page) {
   }
 
   $timeout = _autologout_get_user_timeout();
-  $timeout_padding = \Drupal::config('autologout.settings')->get('autologout_padding');
-  $redirect_url = \Drupal::config('autologout.settings')->get('autologout_redirect_url');
+  $timeout_padding = Drupal::config('autologout.settings')->get('autologout_padding');
+  $redirect_url = Drupal::config('autologout.settings')->get('autologout_redirect_url');
```
Comment (ajitdev) [inline autologout.module:209]: Same. Please revert. For the line above as well.

File: autologout.module [hunk 14]
```diff
@@ -196,19 +205,19 @@ function autologout_page_build(&$page) {
...
   $redirect_query = drupal_get_destination() + array('autologout_timeout' => 1);
-  $no_dialog = \Drupal::config('autologout.settings')->get('autologout_no_dialog');
+  $no_dialog = Drupal::config('autologout.settings')->get('autologout_no_dialog');
```
Comment (ajitdev) [inline autologout.module:211]: Same. Please revert.

File: autologout.module [hunk 15]
```diff
@@ -196,19 +205,19 @@ function autologout_page_build(&$page) {
...
 
   // Get all settings JS will need for dialog.
-  $msg = t('@msg', array('@msg' => \Drupal::config('autologout.settings')->get('autologout_message')));
+  $msg = t('@msg', array('@msg' => Drupal::config('autologout.settings')->get('autologout_message')));
```
Comment (ajitdev) [inline autologout.module:214]: Same. Please revert.

File: autologout.module [hunk 16]
```diff
@@ -196,19 +205,19 @@ function autologout_page_build(&$page) {
...
   $settings = array(
     'timeout' => $refresh_only ? ($timeout * 500) : ($timeout * 1000),
     'timeout_padding' => $timeout_padding * 1000,
     'message' => t('@msg', array('@msg' => $msg)),
     'redirect_url' => url($redirect_url, array('query' => $redirect_query)),
-    'title' => t('@name Alert', array('@name' => \Drupal::config('autologout.settings')->get('site_name'))),
+    'title' => t('@name Alert', array('@name' => Drupal::config('autologout.settings')->get('site_name'))),
```
Comment (ajitdev) [inline autologout.module:220]: Same. Please revert.

File: autologout.module [hunk 17]
```diff
@@ -288,8 +297,8 @@ function _autologout_get_role_timeout() {
 
   // Go through roles, get timeouts for each and return as array.
   foreach ($roles as $rid => $role) {
-    if (\Drupal::config('autologout.settings')->get('autologout_role_' . $rid)) {
-      $timeout_role = \Drupal::config('autologout.settings')->get('autologout_role_' . $rid . '_timeout');
+    if (Drupal::config('autologout.settings')->get('autologout_role_' . $rid)) {
+      $timeout_role = Drupal::config('autologout.settings')->get('autologout_role_' . $rid . '_timeout');
```
Comment (ajitdev) [inline autologout.module:301]: Same. Please revert. For the line above as well.

File: autologout.module [hunk 18]
```diff
@@ -312,7 +321,7 @@ function _autologout_get_user_timeout($uid = NULL) {
 
   if (is_null($uid)) {
     // If $uid is not provided, use the logged in user.
-    $user = \Drupal::currentUser();
+    $user = Drupal::currentUser();
```
Comment (ajitdev) [inline autologout.module:324]: Same. Please revert.

File: autologout.module [hunk 19]
```diff
@@ -323,13 +332,13 @@ function _autologout_get_user_timeout($uid = NULL) {
     return 0;
   }
 
-  if (is_numeric($user_timeout = \Drupal::config('autologout.settings')->get('autologout_user_' . $user->id()))) {
+  if (is_numeric($user_timeout = Drupal::config('autologout.settings')->get('autologout_user_' . $user->id()))) {
```
Comment (ajitdev) [inline autologout.module:335]: Same. Please revert.

File: autologout.module [hunk 20]
```diff
@@ -323,13 +332,13 @@ function _autologout_get_user_timeout($uid = NULL) {
...
     // User timeout takes precedence.
     return $user_timeout;
   }
 
   // Get role timeouts for user.
-  if (\Drupal::config('autologout.settings')->get('autologout_role_logout')) {
+  if (Drupal::config('autologout.settings')->get('autologout_role_logout')) {
```
Comment (ajitdev) [inline autologout.module:341]: Same. Please revert.

File: autologout.module [hunk 21]
```diff
@@ -347,7 +356,7 @@ function _autologout_get_user_timeout($uid = NULL) {
   }
 
   // If no user or role override exists, return the default timeout.
-  return \Drupal::config('autologout.settings')->get('autologout_timeout');
+  return Drupal::config('autologout.settings')->get('autologout_timeout');
```
Comment (ajitdev) [inline autologout.module:359]: Same. Please revert.

File: autologout.module [hunk 22]
```diff
@@ -362,7 +371,8 @@ function _autologout_logout() {
   }
 
   // Destroy the current session.
-  \Drupal::moduleHandler()->invokeAll('user_logout', $user);
+  \Drupal::moduleHandler()->invokeAll('user_logout', array($user));
```
Comment (ajitdev) [inline autologout.module:374]: Same. Please revert.

File: autologout.module [hunk 23]
```diff
@@ -374,9 +384,9 @@ function _autologout_logout() {
  * Helper to determine if a given user should be autologged out.
  */
 function _autologout_logout_role($user) {
-  if (\Drupal::config('autologout.settings')->get('autologout_role_logout')) {
+  if (Drupal::config('autologout.settings')->get('autologout_role_logout')) {
```
Comment (ajitdev) [inline autologout.module:387]: Same. Please revert.

File: autologout.module [hunk 24]
```diff
@@ -374,9 +384,9 @@ function _autologout_logout() {
...
     foreach ($user->roles as $key => $role) {
-      if (\Drupal::config('autologout.settings')->get('autologout_role_' . $key)) {
+      if (Drupal::config('autologout.settings')->get('autologout_role_' . $key)) {
```
Comment (ajitdev) [inline autologout.module:389]: Same. Please revert.

File: autologout.module [hunk 25]
```diff
@@ -390,7 +400,7 @@ function _autologout_logout_role($user) {
  * Display the inactivity message if required.
  */
 function _autologout_inactivity_message() {
-  $message = \Drupal::config('autologout.settings')->get('autologout_inactivity_message');
+  $message = Drupal::config('autologout.settings')->get('autologout_inactivity_message');
```
Comment (ajitdev) [inline autologout.module:403]: Same. Please revert.

File: autologout.module [hunk 26]
```diff
@@ -472,3 +481,46 @@ function autologout_user_login($account) {
       ->execute();
   }
 }
```
Comment (ajitdev) [inline autologout.module:483]: Function name should be followed by a blank line.

File: autologout.module [hunk 27]
```diff
@@ -472,3 +481,46 @@ function autologout_user_login($account) {
...
+/**
+ * Implements hook_page_attachments().
```
Comment (ajitdev) [inline autologout.module:485]: Cool stuff :+1:

File: autologout.module [hunk 28]
```diff
@@ -472,3 +481,46 @@ function autologout_user_login($account) {
...
+ */
+function autologout_page_attachments(array &$page) {
+
+  $refresh_only = _autologout_refresh_only();
+  $now = time();
```
Comment (ajitdev) [inline autologout.module:490]: Use [`REQUEST_TIME`](https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/constant/REQUEST_TIME/8) instead of the `time()` function.

File: autologout.module (hunk 2 above)
Comment (neetumorwani) [inline autologout.module:73]: This will be fixed in next PR.

File: autologout.module (hunk 6 above)
Comment (neetumorwani) [inline autologout.module:75]: Any specific reason for using \Drupal instead of Drupal.
I have included "Drupal" class in the module file. Is there any impact on performance or some other rationale behind this? Just curious to know.
